lib/router/src/cache.rs

```diff
@@ -120,7 +127,10 @@ where
                 .poll()
                 .map_err(|e| unreachable!("expiration must not fail: {}", e)))
             {
-                None => return Ok(Async::NotReady),
+                None => {
+                    self.purge_task = Some(task::current());
```

Comment:
TIOLI: This could use a comment explaining that this is essentially registering the task to be notified when something is added to the delay queue.